bin/index.js

```diff
@@ -24,27 +26,77 @@ const program = require('commander')
     .option('-g, --github [value]', 'add link to Github page')
     .parse(process.argv);
 
-const srcFile = path.join(program.args[0]);
-const destFile = program.output;
-
-program.toc = program.toc || [];
-
-const format = fs.readFileAsync(srcFile, 'utf8')
-    .then(readme => formatter(readme, {
-        lang: program.lang,
-        toc: program.noToc ? false : { minLevel: program.toc.min, maxLevel: program.toc.max },
-        github: program.github
-    }));
-
-if (program.output) {
-    format
-        .then(html => {
-            return makeDir(path.dirname(destFile))
-                .then(() => fs.writeFileAsync(destFile, html))
-        })
-        .catch(err => console.error(err));
-} else {
-    format
-        .then(html => console.log(html))
-        .catch(err => console.log(err));
+let options;
+
+try {
+    // Errors if ToC is given and out of range
+    options = defineSettings(program);
+} catch(err) {
+    failWithError(err);
+}
+
+const destFile = options.output;
+const stdin = process.stdin;
+
+if (destFile) {
+    // Create output directory if it doesn't exist
```

Comment:
doesn't this need a check for relative/absolute path?

Reply:
maybe, but check is similar to what's in the non-stream code. changing this would be a bit beyond the scope of this PR